app/graphql/mutations/test_result/delete.rb

```diff
@@ -14,8 +14,10 @@ class Delete < BaseMutation
 
       def resolve(args = {})
         profile = scoped_profiles.find(args[:profile_id])
-        test_results = scoped_test_results(args).destroy_all
-        profile.destroy! if profile.external
+        policy_profiles = profile.policy_object&.profiles || [profile]
+        test_results = scoped_test_results(profile_id: policy_profiles)
+                       .destroy_all
+        profile.destroy! unless profile.policy_id
```

Comment:
Observation: Only this line seems to fix the described bug.

Reply:
Unfortunately, the list of test results to delete isn't necessarily correct when clicking the 'Delete report' button in the UI for profiles that are part of a policy; thus the other changes to find test results outside of the given profile ID for profiles in a policy.